### DISCUSSION TITLE
Proposal for Phase 8: extending podcast:location

### DISCUSSION THREAD

**Discussion by jamescridland** [Podcastindex-org/podcast-namespace#668] [Enhancement Proposal]
There are a few use-cases that the existing `podcast:location` tag does not deal with, which I'd like to try to fix for the next phase.

The majority of this is the work of Daniel J Lewis; I've extended it yet further (!)

## The current tag (soon to be deprecated)

The current tag is a SINGLE tag and lives in (I think) either the CHANNEL or ITEM.

```xml
<podcast:location geo="geo:30.2672,97.7431" osm="R113314">Austin, TX</podcast:location>
```

## Use case: "I would like to discover podcasts made near me"

The podcast:location tag was initially only to describe what a podcast's *subject* was about, not where it was *recorded*.

We'll fix this by adding a `rel` attribute, explaining what the relationship is between the podcast/episode described and the location; and by allowing this tag to be used multiple times.

* `rel="creator"` is where the podcast was recorded.
* `rel="subject"` is what the podcast is about.

`rel` is a recommended value; in its absence the default value is "subject" to maintain compatibility. However, RSS authors are STRONGLY advised to add this value going forward.

Example: a show that is RECORDED IN Austin, TX - about a themepark in Australia

```xml
<podcast:location geo="geo:30.2672,97.7431" rel="creator" osm="R113314">Austin, TX</podcast:location>
<podcast:location geo="geo:-27.86159,153.3169" rel="subject" osm="W43678282">Dreamworld (Queensland)</podcast:location>
```

## Use case: "I would like a quick and simple method to find podcasts about/from my country"

The podcast:location tag already contains enough information to calculate a country - which can be calculated from the `geo` tag and/or the OSM tag.

However, this is a lot of parsing work for the podcast directory or app; and it would seem to be pragmatic to do that work when _entering the information and generating the RSS feed_, rather than expect an app or client to parse geo/OSM for 4.3mn podcasts; and to avoid multiple clients hitting a geo-lookup or OSM API.

We'll fix this by adding a country code, `country`, which will use the standard two-character country code - ALPHA-2 from ISO 3166.

Ideally, this country code would be mandatory. The question is whether that would actually break anything, or given enough notice, we'd be able to get that done.

An example for a podcast made about The Eiffel Tower might read:

```xml
<podcast:location geo="geo:48.858093,2.294694" osm="W5013364" country="fr" rel="subject">Eiffel Tower, Paris</podcast:location>
```

## Use case: "My podcast is about both Dreamworld *and* the Eiffel Tower"

Sure! Now this tag can be used MULTIPLE times, that's all good.

```xml
<podcast:location geo="geo:48.858093,2.294694" osm="W5013364" country="fr" rel="subject">Eiffel Tower, Paris</podcast:location>
<podcast:location geo="geo:-27.86159,153.3169" rel="subject" osm="W43678282" country="au">Dreamworld (Queensland)</podcast:location>
```

## Use case: "This is all too complicated, James, can't you make it simpler?"

Sure we can.

[This is a JavaScript podcast:location tag generator](https://jamescridland.github.io/podcast-location-generator/).

## Linking to a map

You *can* link to the lat/lon in whatever you'd like (Google, Apple, etc). But the data is so much richer if you link to the actual object - and in some cases, less misleading

If an OSM ID exists, link to it at 
[https://www.openstreetmap.org/relation|way|node/](https://www.openstreetmap.org/relation/)(id)

So, I'm currently in...
```
<podcast:location rel="creator" geo="geo:-27.4689682,153.0234991" osm="R11677792" country="AU">Brisbane</podcast:location>
```
...which, as you'll see from [the OpenStreetMap link](https://www.openstreetmap.org/relation/11677792), doesn't reveal where I live, but does communicate that I'm in a large city. In this case, the lat/lon is misleading, since I'm not sitting in Brisbane City Hall.

## Example feed

https://podnews.net/rss contains a default tag in the `channel` and a number of overwritten tags in the `item` tags, all for rel="creator", as of mid November 2024.

**Comment by samsethi** [comment]
Hi James 

Adding a location service to copy/paste the location is cool.  All hosts etc could use it.  However does this mean we are agreed on the syntax for the updated location tag to be a rel=subject or rel=origin not sure why origin and instead of recorded but ok. 

TrueFans would be happy to test this asap.  Also we plan to add it to a map to visualise the podcasts.

**Reply by jamescridland** [reply]
Yeah, I'm not fond of `"origin"`.

But - if I travel to the Eiffel Tower and record an interview there, but my podcast is based in London, then I as the podcaster am still based in London, irrespective of where I happened to record the show.

Does `"author"` work better? `"creator"` ?

**Reply by samsethi** [reply]
When we were only working with podcasters then author was easy. But now we have musicians and thus we started to use the generic term creators to cover both podcasters and artists.

**Reply by jamescridland** [reply]
I've edited the proposal to use `creator`.

**Comment by daveajones** [comment]
I love everything about this.  Thanks for doing this work @jamescridland and Daniel

**Comment by Marzal** [comment]
Great potential of this idea. So I added the changes to a feed in case someone wants to play with it:
Channel -> Location `creator`
EP [17](https://truefans.fm/kde-express/65d074dc3ebce3f7eae06b62) and 27 -> Location `subject`
https://kdeexpress.gitlab.io/feed

@samsethi I agree with James, about a different icon for creator/subject (I would also add a tool-tip)
https://truefans.fm/kde-express/65d074dc3ebce3f7eae06b62
Also if `osm` is on the tag, another icon to https://www.openstreetmap.org/... would be great, using `geo` as a fallback
So I would put 2 links, OSM location if `osm` exists and Gmaps/geohack [3] if only `name/geo` is available. Maybe a link in the icon an another in the locations's name.

PD: This could be useful: find places [1] , link to OSM with the tag info [2], 
[1] https://wiki.openstreetmap.org/wiki/Nominatim
[2] https://wiki.openstreetmap.org/wiki/Browsing#Other_URL_tricks
[3] https://geohack.toolforge.org/geohack.php?params=39.4585_N_-0.33766666666666667_E_globe:earth&language=en

**Reply by samsethi** [reply]
Thanks David and agree will look at how we can best show this in the UI.  The Tooltip is an easy quick win.

**Comment by jamescridland** [comment]
For those wishing to test this, Podnews's RSS feed - https://podnews.net/rss - now has compliant location tags in both the `channel` and `item` sections. Data is correct from mid-October - note that where no location is given in the `item`, you default back to the `channel`definition. Podnews only uses `rel="creator"` and not the subject.

**Reply by samsethi** [reply]
We have taken on board your fallback. And pushing that update today. 


We have implemented the new location tag as originally proposed by Daniel J Lewis and refined by James Cridland. Below are four recent episodes from Podnews Daily showing James' different recording locations.  e.g https://truefans.fm/podnews-daily/671f5681d55de476b7cb4d4a

If you click the location tag we show it on a map.  Our next feature is to add multiple podcasts on a map so you can see podcasts nearby and filter by category. 

We have the Cridland feature.  Where we show all the locations of a single podcaster on a map but we can only think James will be the person who records in multiple locations.

<img width="614" alt="Cursor_and_TrueFans___Shows___Podcast_Days_takes_place_in_Madrid_🔊" src="https://github.com/user-attachments/assets/4a477d70-0433-4564-af93-a47bc23e88da">

<img width="681" alt="TrueFans___Shows____It’s_a_Continent__is_Independent_Podcast_of_the_year_🔊" src="https://github.com/user-attachments/assets/eb3d24ad-d45f-4f11-8d44-f6f0ede1828b">

<img width="735" alt="Cursor_and_TrueFans___Shows___The_world’s_largest_podcast_agency_is_created_🔊" src="https://github.com/user-attachments/assets/fd314bc9-9503-4c03-a86f-d48391c93cf5">

<img width="646" alt="TrueFans___Shows___Women_in_Podcasting_Awards_winners_🔊" src="https://github.com/user-attachments/assets/c8fc1647-9452-4c3e-b935-c503e3669ebd">

**Comment by daveajones** [comment]
Do we have any objections to this?  I don’t see why we would delay formalizing these changes.
